### ISSUE TITLE
orientation flips when looking above the horizon

### ISSUE THREAD

**Issue by Aeva** [richtr/Full-Tilt#14]
The WebGL VR demo works fine when I'm holding my phone in the portrait orientation, but when I hold it in horizontal orientation (such as to be used with Google Cardboard or the likes), the orientation is inverted when looking above the horizon.  Looking below the horizon has the orientation as expected.  The phone's auto rotate feature is turned off.

Tested this on Samsung Galaxy S6 with Firefox Beta.

I'm interested in using this for a VR project, and would be happy to help with a patch.  Any idea why this is happening?

**Comment by richtr** [comment]
Firefox for Android currently does something strange with the `gamma` value that it returns as part of the [Device Orientation Events API](https://w3c.github.io/deviceorientation/spec-source-orientation.html). It does not match the `gamma` values returned in e.g. Chromium-based Android browsers.

Most uses of device orientation rely on consistency between device orientation data between user agents. This library already manages to work around differences between iOS and Chromium-based Android browsers emitting different device orientation data (it just happens that these devices coincidentally provide _mathematically equivalent_ device orientation data to each other).

The only potential fix right now would be to sniff for the Firefox for Android UA string and fix the `gamma` value in to something we expect for use in Full-Tilt. 

I would prefer for Mozilla to fix this issue directly in the browser though. I don't want to start introducing hacks in to this library because once it gets fixed it will break any existing Full-Tilt usage currently deployed elsewhere on the web.

Please consider filing a bug on [Bugzilla@Mozilla](https://bugzilla.mozilla.org/describecomponents.cgi?product=Firefox%20for%20Android) for Firefox for Android. If you do file a bug please paste the URL here so we can track further progress on this.

**Comment by Aeva** [comment]
at least one such bug already exists in their tracker https://bugzilla.mozilla.org/show_bug.cgi?id=1176801

**Comment by richtr** [comment]
Yep, that looks like the right bug. I hope they will fix it soon!
